modules/ROOT/pages/clustering/clustering-advanced/reconciler.adoc

```diff
@@ -0,0 +1,58 @@
+[role=enterprise-edition]
+[[cluster-reconciler]]
+= Reconciler
+:description: This section describes how changes to the DBMS are processed by each server.
+
+[[cluster-reconciler-introduction]]
+== Introduction
+
+Administrative operations in Neo4j, such as creating a database, do not happen synchronously.
+Instead we have a model where the changes are written to the `system` database, and record the desired state of the DBMS.
+Each server has an internal component called the reconciler, which observes the desired state and makes changes to the local server to match that state.
+For example, the `system` database may record that it wants the database `bar` running on `server-1`.
+When the reconciler on `server-1` sees that, it will start `bar`.
+
+Executing an operation and getting back a successful response means that the desire is safely committed to the `system` database, and will be processed by each member of the cluster at some point, assuming the server is healthy enough.
+
+[[cluster-reconciler-async]]
+== Asynchronicity
+Servers can observe the same operation at different times, since the `system` database is just another Raft group, where followers and replicas can lag behind the leader in some situations.
+Each server should eventually see the new state though, and take action on it.
+
+[[cluster-reconciler-wait]]
+== Waiting
+
+If you want to be sure that each server has processed an administrative operation before you run the next action, you can use the `WAIT` keyword.
+`WAIT` makes the statement not return until all servers have seen the transaction, and their reconciler has finished processing it.
+
+[NOTE]
+====
+A statements with `WAIT` returning does not mean that your operation is necessarily successful everywhere.
+`WAIT` just ensures the operation has been processed.
+A request to start a database has still been processed if the reconciler tried to start it and the database failed and went into the dirty state.
+====
+
+=== Example using `WAIT`
+[source, cypher, role="noplay"]
+----
+CREATE DATABASE foo WAIT
+----
+
+[[cluster-reconciler-errors]]
+== Errors
+
+An operation might only succeed on some servers.
+For instance some servers might be offline when you create a database, or a disk error might cause a server to fail to start a database.
+In this situation the operation has not 'failed' as a whole, since you may even have a running database, just with less fault tolerance than you desired.
+
+Some failures can just be re-attempted, for example `START DATABASE` can be run if not all members started successfully, and you think the reason has been resolved.
+Other failures will need just the underlying problem to be resolved and the reconciler will make the change, because it is still trying to achieve the desired state.
```

Comment:
```suggestion
Other failures only require the underlying problem to be resolved. 
When the issue is fixed, the reconciler will make the intended change, because it is still trying to achieve the desired state.
```